Shared/Samples/Snap geometry edits with utility network rules/SnapGeometryEditsWithUtilityNetworkRulesView.swift

```diff
@@ -0,0 +1,280 @@
+        case editing
+        /// The feature's edits are being saved to its table.
+        case saving
+    }
+    
+    /// The current state of the sample.
+    @State private var state = SampleState.setup
+    
+    /// A Boolean value indicating whether there are edits to be saved.
+    @State private var canSave = false
+    
+    /// A Boolean value indicating whether snap sources list is showing.
+    @State private var sourcesListIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The instruction text indicating the next step in the sample's workflow.
+    private var instructionText: String {
+        if state == .editing {
+            canSave
+            ? "Tap the save button to save the edits."
+            : "Tap on the map to update the feature's geometry."
+        } else {
+            model.selectedElement == nil
+            ? "Tap the map to select a feature."
+            : "Tap the edit button to update the feature."
+        }
+    }
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: model.map, graphicsOverlays: [model.graphicsOverlay])
+                .geometryEditor(model.geometryEditor)
+                .onSingleTapGesture { screenPoint, _ in
+                    state = .identifying(tapPoint: screenPoint)
+                }
+                .task(id: state) {
+                    // Runs the async action related to the current sample state.
+                    do {
+                        switch state {
+                        case .setup:
+                            try await model.setUp()
+                        case .identifying(let tapPoint):
+                            let identifyResults = try await mapViewProxy.identifyLayers(
+                                screenPoint: tapPoint,
+                                tolerance: 5
+                            )
+                            try await model.selectFeature(from: identifyResults)
+                        case .editing:
+                            model.startEditing()
+                            
+                            for await canUndo in model.geometryEditor.$canUndo {
+                                canSave = canUndo
+                            }
+                        case .saving:
+                            try await model.save()
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .errorAlert(presentingError: $error)
+        }
+        .overlay(alignment: .top) {
+            VStack(alignment: .trailing, spacing: 0) {
+                Text(instructionText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.ultraThinMaterial, ignoresSafeAreaEdges: .horizontal)
```

Comment:
`ultraThinMaterial` is pretty dark. Consider using `thinMaterial` for all overlays.